lib/src/command.dart

```diff
@@ -110,6 +111,8 @@ abstract class PubCommand extends Command<int> {
     return _pubEmbeddableCommand ?? (runner as PubCommandRunner);
   }
 
+  Analytics get analytics => _pubEmbeddableCommand?.analytics;
```

Comment:
How about we make a `_NullAnalytics` and/or a `_LoggingAnalytics` implementation that just logs everything ping with log-level `fine`, I think that's useful for debugging, testing and whenever analytics is disabled.